### PR TITLE
Add use and documentation of west manifest to mcuboot

### PR DESCRIPTION
**Pull request by sigvartmh**
This PR will add a west manifest and remove unused symbols used for building cc310 support for nRF52840 in MCUBoot. Now mcuboot can use west to initialize zephyr it can also use revisions in the manifest to lock the support to a specific revision of zephyr.

### CODE REVIEW

**Comment by sigvartmh** [comment]
@utzig @nvlsianpu

**Comment by sigvartmh** [comment]
Couldn't find a way to put `west.yml` in `boot/zephyr` so I hope having it in the root folder is okei.

**Comment by nvlsianpu** [comment]
@sigvartmh I think we already stated that west manifest should be provided by parent project like fw-nrfconnect-nrf. I something form this PR still needed to be integrated to the mcuboot?

File: west.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+west:
```
Comment (nvlsianpu) [inline west.yml]: file TBR

Comment (sigvartmh) [reply west.yml]: If the maintainers of MCUBoot will allow tight coupling with fw-nrfconnect-nrf I'm fine with that and I'll change this PR

Comment (sigvartmh) [reply west.yml]: @nvlsianpu: This is now updated

**Pull request closed without merging**